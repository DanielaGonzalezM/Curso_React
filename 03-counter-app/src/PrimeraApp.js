//import { Fragment } from "react";
import PropTypes from 'prop-types';

const PrimeraApp = ({ pregunta, subtitulo}) => {

    const saludo = 'Hola ';
    const persona = {
        nombre: 'Daniela',
        edad: 30
    }


    return (
        // <Fragment> tambien sirve en vez de <>
        <>
            <h1>{saludo} {JSON.stringify(persona.nombre)}</h1>
            {/* <p>{pregunta}</p> */}
            {/* <pre>JSON:{JSON.stringify(persona,null,3)}</pre> */}
            {/* <p>Mi primera Aplicaci√≥n</p> */}
            <p>{subtitulo}</p>
        </>
        // </Fragment>


    );

}

PrimeraApp.propTypes = {
    pregunta: PropTypes.string.isRequired
}

PrimeraApp.defaultProps={
    subtitulo: 'Soy un subtitulo' 
}
export default PrimeraApp;    
