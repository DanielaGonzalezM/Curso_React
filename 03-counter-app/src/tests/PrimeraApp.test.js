import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme';

describe('Pruebas en PrimeraApp', () => {
    // test('Debe de mostrar el mensaje como estas?', () => { 
    //     const saludo ="como estas?";
    //     const {getByText} = render(<PrimeraApp pregunta={saludo}/>);
    //     expect(getByText(saludo)).toBeInTheDocument();
    //  });
    test('debe mostrar PrimeraApp correctamente', () => {
        const pregunta = "como estas?";
        const wrapper = shallow(<PrimeraApp pregunta={pregunta} />);

        expect(wrapper).toMatchSnapshot();
    });


    test('debe mostrar el subtitulos enviado por props', () => {
        const pregunta = "como estas?";
        const subtitulo = 'Soy un subtitulo';
        const wrapper = shallow(<PrimeraApp pregunta={pregunta} subtitulo={subtitulo} />);
        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo);

    });

});