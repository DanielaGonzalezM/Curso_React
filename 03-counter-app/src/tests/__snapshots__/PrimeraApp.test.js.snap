// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp debe mostrar PrimeraApp correctamente 1`] = `
<Fragment>
  <h1>
    Hola 
     
    "Daniela"
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
